src/features/cascade-delete/hooks.ts

```diff
@@ -16,13 +15,97 @@ setupDeleteCascade('application', [
 	['application', 'depends_on__application'],
 ]);
 
+hooks.addPureHook('DELETE', 'resin', 'application', {
+	/**
+	 * Cascade from app -> device when deleting the app to match pre-single-multi-app behavior
```

Comment:
Will proper multi-app permit devices with no apps? If not, you could delete the device if it was only associated with the application the request is deleting. You may want to add a feature-specific TODO here either way as a reminder.

Reply:
Yes it definitely allows it, a device will be part of a fleet and a fleet can have as many or as few apps as it wants - and a fleet with no apps is essentially a "waiting room"/"staging area"